libs/components/datetime/src/lib/modules/date-range-picker/date-range-picker.component.ts

```diff
@@ -273,27 +275,33 @@ export class SkyDateRangePickerComponent
 
       this.isReady = true;
 
-      this.resetFormGroupValue();
       this.showRelevantFormFields();
 
-      // Fill in any unprovided values after the calculators have been initialized.
-      // For example, if the control is initialized with only the `calculatorId`,
-      // allow the calculator to fill in the missing start and end dates.
-      const { startDate, endDate } = this.value;
-      const defaultValue = this.selectedCalculator.getValue(startDate, endDate);
-      const newValue = Object.assign({}, defaultValue, this.value);
-
-      this.setValue(newValue, false);
-
-      // This is needed to address a bug in Angular 4.
-      // When a control value is set intially, its value is not represented on the view.
-      // See: https://github.com/angular/angular/issues/13792
-      /* istanbul ignore else */
-      if (this.control) {
-        this.control.setValue(this.value, {
-          emitEvent: false,
-        });
-      }
+      this.ngZone.onStable.pipe(first()).subscribe(() => {
```

Comment:
Why do we need to use `ngZone`? Mind adding a comment above for future reference?

Reply:
Added a comment. Using `ngZone` as it is more reliable than `setTimeout`. Alex and I recently used a similar techinique in Flyout based on what Angular CDK does under the hood for focus stuff.